### PR TITLE
Increase memory limit for ROSA/HCP env

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
In ROSA/HCP env, the gatekeeper-audit pod failed with OOMKilled. 
This PR increases the memory limit to 1Gi to fix this issue. 

Ref: https://issues.redhat.com/browse/ACM-9757

### CODE REVIEW

File: controllers/gatekeeper_controller_test.go [hunk 1]
```diff
@@ -557,32 +557,32 @@ func TestResources(t *testing.T) {
...
 
 	gatekeeper.Spec.Webhook = webhook
 	err = crOverrides(gatekeeper, WebhookFile, webhookObj, namespace, false, false)
 	g.Expect(err).ToNot(HaveOccurred())
-	assertResources(g, webhookObj, webhook.Resources)
+	assertResources(g, webhookObj, webhook.Resources, false)
 }
 
-func assertResources(g *WithT, obj *unstructured.Unstructured, expected *corev1.ResourceRequirements) {
+func assertResources(g *WithT, obj *unstructured.Unstructured, expected *corev1.ResourceRequirements, isAudit bool) {
```
Comment (dhaiducek) [inline controllers/gatekeeper_controller_test.go:585]: The `isAudit` boolean is unnecessary--you should just pass in the `expected` for all audit tests and leave the `assertResources` function as it is, right?

File: config/gatekeeper/kustomization.yaml [hunk 2]
```diff
@@ -43,6 +43,13 @@ patches:
   target:
     kind: Deployment
     name: gatekeeper-audit
+- patch: |-
+    - op: replace
+      path: /spec/template/spec/containers/0/resources/limits/memory
+      value: 1Gi
+  target:
+    kind: Deployment
+    name: gatekeeper-audit
```
Comment (dhaiducek) [inline config/gatekeeper/kustomization.yaml:52]: All the patches above this one target the `gatekeeper-audit` Deployment. Can we consolidate them into a single `patch` array?

**Review by dhaiducek — CHANGES_REQUESTED** [review]
Thanks for the quick update! Looks good--I have a couple questions/comments.

File: config/gatekeeper/kustomization.yaml [hunk 3]
```diff
@@ -30,16 +30,15 @@ resources:
 # Remove --disable-cert-rotation
 # Set a CPU limit
```
Comment (dhaiducek) [inline config/gatekeeper/kustomization.yaml:31]:
```suggestion
# Set a CPU limit
# Increase default Memory limit
```

**Review by dhaiducek — CHANGES_REQUESTED** [review]
Looks way cleaner--thanks!

Please add the Jira issue to the commit message, and this should be good to go!

**Review by dhaiducek — APPROVED** [review]

**Comment by yiraeChristineKim** [comment]
@dhaiducek  ooooh!! Thanks for cherrypicking

**Pull request merged**